### PR TITLE
Add unordered case for TransitionChoice + use bins

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/parametrization/choice.py [hunk 1]
```diff
@@ -170,6 +171,8 @@ class TransitionChoice(BaseChoice):
         the transition weights. During transition, the direction (forward or backward will be drawn with
         equal probabilities), then the transitions weights are normalized through softmax, the 1st value gives
         the probability to remain in the same state, the second to move one step (backward or forward) and so on.
+    ordered: bool
+        change defaults behavior to be unordered and sampled uniformly through a Gaussian in the data space.
```
Comment (jrapin) [inline nevergrad/parametrization/choice.py:175]:
```suggestion
        if False, changes defaults behavior to be unordered and sampled uniformly through a Gaussian in the data space.
```

File: nevergrad/parametrization/choice.py [hunk 2]
```diff
@@ -158,7 +158,8 @@ def mutate(self) -> None:
 
 
 class TransitionChoice(BaseChoice):
-    """Ordered categorical parameter, choosing one of the provided choice options as a value, with continuous transitions.
+    """Categorical parameter, choosing one of the provided choice options as a value, with continuous transitions.
+    By default, this is ordered, and most algorithms except OnePlusOne based algorithms will consider it as ordered.
```
Comment (teytaud) [inline nevergrad/parametrization/choice.py:162]: Isn't it "DiscreteOnePlusOne" rather than OnePlusOne ? I would expect OnePlusOne to keep the ordered style.

**Comment by teytaud** [comment]
I think that in pbo_suite we need three cases:
- "Softmax" which leads to the logit stuff as we know it.
- "Unordered" which leads to TransitionChoice and ordered=False
- "Ordered" which leads to TransitionChoice and ordered=True
so that results will be very cool.

It's:
(i) adding something like:
<<
        elif instrumentation == "Unordered":
            parameterization = ng.p.TransitionChoice([0, 1], repetitions=dim)
>> in https://github.com/facebookresearch/nevergrad/blob/21ac3f4570127a9946cdb45e60c0245a3dddec35/nevergrad/functions/iohprofiler/core.py#L57
and (ii) adding "Ordered" in
https://github.com/facebookresearch/nevergrad/blob/21ac3f4570127a9946cdb45e60c0245a3dddec35/nevergrad/functions/iohprofiler/core.py#L52
and (iii) adding "Ordered" in
https://github.com/facebookresearch/nevergrad/blob/21ac3f4570127a9946cdb45e60c0245a3dddec35/nevergrad/benchmark/experiments.py#L1928

and the result of this experiment will be very interesting (I can run that if you wish, I do this kind of run everyday :-) ).

**Review by teytaud — APPROVED** [review]

**Review by mathuvu — APPROVED** [review]
